rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -96,27 +106,28 @@ get_node_names(
 
       if (name_found != map.end()) {
         name = std::string(name_found->second.begin(), name_found->second.end());
-      }
-      if (name_found != map.end()) {
         namespace_ = std::string(ns_found->second.begin(), ns_found->second.end());
       }
+
+      // ignore discovered participants without a name
       if (name.empty()) {
         // use participant name if no name was found in the user data
         if (pbtd.participant_name.name) {
           name = pbtd.participant_name.name;
         }
       }
     }
+
+    // ignore discovered participants without a name
     if (name.empty()) {
-      // ignore discovered participants without a name
-      node_names->data[i] = nullptr;
+      tmp_names_list.data[i] = nullptr;
       node_namespaces->data[i] = nullptr;
       continue;
     }
 
-    node_names->data[i] = rcutils_strdup(name.c_str(), allocator);
-    if (!node_names->data[i]) {
-      RMW_SET_ERROR_MSG("could not allocate memory for node name");
+    tmp_names_list.data[named_nodes_num] = rcutils_strdup(name.c_str(), allocator);
```

Comment:
I think this should use `i`, as we're using it on the following line.
```suggestion
    tmp_names_list.data[i] = rcutils_strdup(name.c_str(), allocator);
```

Reply:
I guess that that's not right. My idea was to overallocate `tmp_names_list`, so it might be identical to `node_names`. Next, I add node names to tmp_names_list only if a name is not empty. Thus, `named_nodes_num` tracks number of non-empty names so far. After traversing the whole list of names, `tmp_names_list` contains valid names and (possibly) nullptrs at the end. We can safely truncate them later. Is this logic applicable?

Reply:
Okay, makes sense. We should do the same logic for the namespaces as well, no?

Reply:
I think so. This function treats names and namespaces in almost identical way.

Reply:
Done